### ISSUE TITLE
Amp-analytics substitutions do not work for array vars

### ISSUE THREAD

**Issue by samdozor** [ampproject/amphtml#21224]
## What's the issue?

When specifying an array as an amp-analytics vars, variable substitutions are not performed. Array support in custom vars was added in issue https://github.com/ampproject/amphtml/issues/1886, but it seems this aspect of it wasn't implemented.

I believe this affects both custom variable substitution and the built-in variables. However, for built in variables I believe you can workaround this issue by using the platform variable (eg. `AMPDOC_HOST`) rather than the amp-analytics variable (eg. `${ampdocHost}`)

### JSBin Link
https://jsbin.com/xoyuqob/edit?html,output

### Steps to Reproduce

1. Create a link with a "data-vars-" attribute and value, for example:
    ```html
    <a href="#" data-vars-custom-event-attr="foo-value" 
                data-vars-custom-event-name="foo custom event name">Example</a>
    ```
2. Attempt to use the data var `${customEventAttr}` in an array for an amp-analytics var:
   ```js
    "triggers": {
          "trackAnchorClicks": {
               "on": "click",
               "selector": "a",
               "request": "event",
               "vars": {
                   "eventAttributes_Keys": ["foo", "bar"],
                   "eventAttributes_Values": ["${customEventAttr}","bar-value"],
                   "eventName": "${customEventName}"
               }
          }
    }
   ```
3. Click on the link. 
4. Observed Behavior:
    - the generated pixel URL does not replace "customEventAttr" with "**foo-value**". 
    - Example: https://pixels.mparticle.com/v1/foobar/Pixel?dt=AppEvent&n=foo%20custom%20event%20name&et=Unknown&amp_id=amp-CCQgZ6lmWuvudklhCemUAg&attrs_k=foo,bar&attrs_v=**%24%7BcustomEventAttr%7D,bar-value**&ua_k=&ua_v=&ui_t=&ui_v=&flags_k=&flags_v=&ct=1551668866036&dbg=false&lc=&av=

5. Expected behavior:
   - the generated pixel URL should replace each individual array item with any custom vars
   - Example: https://pixels.mparticle.com/v1/foobar/Pixel?dt=AppEvent&n=foo%20custom%20event%20name&et=Unknown&amp_id=amp-CCQgZ6lmWuvudklhCemUAg&attrs_k=foo,bar&attrs_v=**foo-value,bar-value**&ua_k=&ua_v=&ui_t=&ui_v=&flags_k=&flags_v=&ct=1551668866036&dbg=false&lc=&av=

## What browsers are affected?

All browsers.

## Which AMP version is affected?

Always broken.

**Comment by torch2424** [comment]
Tested the provided example, and confirmed the issue:

![screen shot 2019-03-07 at 6 15 58 pm](https://user-images.githubusercontent.com/1448289/54002999-74b26100-4105-11e9-8d90-5a177a858832.png)


cc @lannka @zhouyx

**Comment by samdozor** [comment]
@torch2424 shall I submit a PR for this puppy or do you know if the team was already considering it in their backlog?

**Comment by zhouyx** [comment]
Thank you @samdozor for reporting and thank you @torch2424 for reproducing. I've been able to verify as well. 

Looks like it's related to 
https://github.com/ampproject/amphtml/blob/313def74938be9b264daf0b0986c98fa17c19bf2/extensions/amp-analytics/0.1/variables.js#L218-L232

Since this is not a popular request, we haven't prioritized this request. It would great if you can contribute. Thank you!

@calebcordry I believe you're most familiar with the expander code. Could you please help here. Thank you!

**Comment by calebcordry** [comment]
Sure, happy to help. Looks like you have already found the code pointers. We could maybe just stringify the array before calling `expandTemplateSync()` I would have to look at the code a bit more to be sure, I can do this when reviewing, or it gets prioritized. Thanks!

**Comment by micajuine-ho** [comment]
Closed via #26157
